source/language/types.rst

```diff
@@ -210,24 +210,51 @@ compatible with run-time values on some platforms.
    // Declare a machine-sized angle
    angle my_machine_angle;
 
+
 Complex numbers
 ~~~~~~~~~~~~~~~
 
-Complex numbers may be declared as ``complex[type] name``, for a numeric OpenQASM classical type
-``type`` (``int``, ``float``, ``angle`` plus their optional size). The real
-and imaginary parts of the complex number are ``type[size]`` types. For instance, ``complex[float[32]] c``
-would declare a complex number with real and imaginary parts that are 32-bit floating point numbers. The
-``im`` keyword defines the imaginary number :math:`sqrt(-1)`. ``complex[type[size]]`` types are initalized as
-``a + b im``, where ``a`` and ``b`` must be of the same type as ``type[size]``. ``b`` must occur to the
-left of ``im`` and the two can only be seperated by spaces/tabs (or nothing at all).
+Complex numbers may be declared as ``complex[float[size]] name``, where ``size``
+is the size of the IEEE-754 floating-point number used to store the real and
+imaginary components.  Each component behaves as a ``float[size]`` type.  The
+designator ``[size]`` can be omitted to use the default hardware ``float``.
```

Comment:
Should we go further and say that if `float` is omitted then it is assumed? ie. `complex` == `complex[float]`

Reply:
I am okay with that.

Reply:
We'd originally left this in deliberately as a nod towards Blake and Prasahnt's potential extension in a later language spec.  I'm happy to change it if Blake's happy.

Changed in b3caee1.